Code/GraphMol/RWMol.cpp

```diff
@@ -119,6 +123,10 @@ unsigned int RWMol::addAtom(bool updateLabel) {
     clearAtomBookmark(ci_RIGHTMOST_ATOM);
     setAtomBookmark(atom_p, ci_RIGHTMOST_ATOM);
   }
+
+  // SGroups do not tolerate modification of the molecule, so drop them
+  clearSGroups();
```

Comment:
I doubt that sgroups will fail when an atom is added.  None of the other atom pointers or indices change.

Reply:
In this case, and the ones in the next comments, I wanted to be on the safe side: maybe from a programming point of view adding atoms/bonds/molecules won't cause problems, but I thought that from a chemical point the addition of any of those might change/invalidate the information stored in the SGroups, so it might be better to completely remove them.

Reply:
Of course, this is something I am happy to discuss about, and I know that it will most probably change in the close future.